### PR TITLE
Add more peripherals for the stm32f030

### PR DESCRIPTION
**Pull request by david-sawatzke**
- Serial is now structured like in https://github.com/stm32-rs/stm32f4xx-hal/pull/48/, so there is no giant macro repeating code
- I2C1 & I2C2 is there
- SPI1 & SPI2 is there

### TODO
- (Maybe) Port examples so they work on all variants (Opened a separate issue for that)

### CODE REVIEW

File: src/serial.rs [hunk 1]
```diff
@@ -83,63 +83,83 @@ pub struct Serial<USART, PINS> {
...
+    // USART6: (usart6, usart6en, apb2enr),
+}
+
+// It's s needed for the impls, but rustc doesn't recognize that
+#[allow(dead_code)]
+type SerialRegisterBlock = stm32::usart1::RegisterBlock;
+
+impl<USART> embedded_hal::serial::Read<u8> for Rx<USART>
+where
+    USART: Deref<Target = SerialRegisterBlock>,
```
Comment (HarkonenBade) [inline src/serial.rs]: This again feels kinda sketchy to me, as it is relying on all the USART peripherals continuing to use the same register block definition.

Comment (david-sawatzke) [reply src/serial.rs]: If it's going to break, we're getting a compile error, so in that's ok. In the worst case you can do some macro based approach where you can separately implement the different register blocks.

**Comment by david-sawatzke** [comment]
Added all possible pin combinations and split up the pin traits for Serial

File: src/serial.rs [hunk 2]
```diff
@@ -61,49 +61,55 @@ pub enum Error {
...
-impl Pins<stm32::USART2> for (gpioa::PA14<Alternate<AF1>>, gpioa::PA3<Alternate<AF1>>) {}
+usart_pins! {
+    USART2: (gpioa::PA2<Alternate<AF1>>, gpioa::PA3<Alternate<AF1>>),
+    USART2: (gpioa::PA14<Alternate<AF1>>, gpioa::PA15<Alternate<AF1>>),
+}
+#[cfg(feature = "stm32f030xc")]
+usart_pins! {
+    // TODO WTF look at this again, in the datasheet PB10 is both tx and rx
+    // USART3: (gpiob::PB10, AF4, gpiob::PA11, AF4),
+    USART3: (gpioc::PC4<Alternate<AF1>>, gpioc::PC5<Alternate<AF1>>),
```
Comment (therealprof) [inline src/serial.rs]: Instead of specifying pairs wouldn't it make more sense to actually single out the TX/RX functions and let the user chose any possible combination? Using this macro in this way means you're passing on a few possible combinations, e.g. you use also use PC10 and PC5 for USART3, or PC4 and PC11.

Comment (david-sawatzke) [reply src/serial.rs]: Thats the whole purpose of the separate Rx and Tx traits, the combinations are done by the templates. The only reason i defined them pairwise is because that what ST seems to have done and it seemed cleaner this way.

Comment (therealprof) [reply src/serial.rs]: @HarkonenBade That sounds like a good idea.

Comment (david-sawatzke) [reply src/serial.rs]: @HarkonenBade Agree. We could also reuse that for the spi, where the pin combinations aren't that clear.

Comment (HarkonenBade) [reply src/serial.rs]: Done, if @david-sawatzke could pull the fresh commit from https://github.com/HarkonenBade/stm32f0xx-hal/tree/stm32f030_v2 that would be great.

Comment (HarkonenBade) [reply src/serial.rs]: I took some slight liberties with making the macro syntax a a bit more verbose, but I think it makes it a lot clearer what is going on.

Comment (david-sawatzke) [reply src/serial.rs]: @HarkonenBade Yeah, that's better. The syntax also seems more "rusty".

**Comment by therealprof** [comment]
Is this still WIP? Looks pretty much complete to me.

**Comment by david-sawatzke** [comment]
I'd also like to port spi & i2c to the same approach as serial and add the 2nd instance

**Comment by david-sawatzke** [comment]
Currently getting a rustc panic
```
thread 'main' panicked at 'byte index 1177 is out of bounds of `crate::stm32::$SPI`', libcore/str/mod.rs:2094:9
note: Run with `RUST_BACKTRACE=1` for a backtrace.

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

```

File: Cargo.toml [hunk 3]
```diff
@@ -51,8 +51,8 @@ stm32f042 = ["stm32f0/stm32f0x2"]
 stm32f030 = ["stm32f0/stm32f0x0"]
 stm32f030x4 = ["stm32f030x6"]
 stm32f030x6 = ["stm32f030"]
-stm32f030x8 = ["stm32f030", "stm32f0/stm32f0x0"]
-stm32f030xc = ["stm32f030", "stm32f0/stm32f0x0"]
+stm32f030x8 = ["stm32f030"]
+stm32f030xc = ["stm32f030"]
```
Comment (therealprof) [inline Cargo.toml]: Snuck in another unrelated change, eh? ;)

Comment (david-sawatzke) [reply Cargo.toml]: It's technically not *completely* unrelated
